gcloud-java-storage/src/main/java/com/google/gcloud/storage/StorageOptions.java

```diff
@@ -34,12 +35,47 @@
   private static final String DEFAULT_PATH_DELIMITER = "/";
 
   private final String pathDelimiter;
+  private final TimeSource timeSource;
   private transient StorageRpc storageRpc;
 
+  /**
+   * A class providing access to the current time in milliseconds.
+   *
+   * Implementations should implement {@code Serializable} wherever possible and must document
+   * whether or not they do support serialization.
+   */
+  public static abstract class TimeSource {
```

Comment:
The main point for not putting it serializable is that Java8's Clock is not serializable so I did not want to introduce something that we'll need to change when adopting Java8's Clock. I wouldn't mind changing anyway.